app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go

```diff
@@ -214,28 +213,30 @@ func (openAiCodeGenerator OpenAiNextJsCodeGenerator) Execute(step steps.Generate
 func (openAiCodeGenerator *OpenAiNextJsCodeGenerator) buildFinalInstructionForGeneration(
 	step steps.GenerateCodeStep, storyDir string) (map[string]string, error) {
 	// Initialize the final instruction string
-	finalInstruction, err := openAiCodeGenerator.buildInstructionForFirstExecution(step, storyDir)
-	if err != nil {
-		fmt.Printf("Error building instruction for first execution: %s\n", err.Error())
-		return nil, err
-	}
+	var finalInstruction map[string]string
 	if step.Retry {
 		fmt.Println("Building instruction on retry limit reached for LLM steps")
+		var err error
 		finalInstruction, err = openAiCodeGenerator.buildInstructionOnRetry(step, storyDir)
 		if err != nil {
 			fmt.Printf("Error building instruction on retry: %s\n", err.Error())
 			return nil, err
 		}
 	} else if step.Execution.ReExecution {
+		var err error
```

Comment:
I meant `finalInstruction, err := openAiCodeGenerator.buildInstructionOnReExecutionWithComments(step, storyDir)` should work right?